ui/segmented-bar.reel/segmented-bar.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * @module ui/segmented-bar.reel
+ * @requires montage/ui/component
+ */
+var Component = require("montage/ui/component").Component;
+
+/**
+ * @class SegmentedBar
+ * @extends Component
+ * A SegmentedBar is a horizontal/vertical control made of multiple segments, each
+ * segment functioning as a discrete button. The segments clicked reflects the data
+ * associated with the segment. The segment data will be displayed in a separate panel.
+ * A percentage of the SegmentedBar will reflect unused space. The segments will draw
+ * themselves depending on the percentage of the total length of the bar. The SegmentedBar
+ * will accept an array of data or a contentController('rangeController' / 'treeController').
+ */
+exports.SegmentedBar = Component.specialize({
+    _content: {value: null},
+    /**
+     * Accept `[]`
+     * content data accepts an array of objects defined within the properties
+     * as a part of the serialization inside the component that contains
+     * the SegmentedBar.
+     */
+    content: {
+        get: function () {return this._content},
+        set: function (content) {
+            this._content = content;
+        }
+    },
+    /**
+     * contentController accepts a 'RangeController' or 'TreeController', the
+     * contentController will default to a RangeController. The TreeController
+     * serves to flatten out a tree to be portrayed as a bar.
+     */
+    contentController: {value: null},
+
+    labelKeyName: {value: null},
+    valueKeyName: {value: null},
+    remainingLabelName: {value: null},
+    remainingValue: {value: null},
+
+    _maxValue: {value: null},
+    maxValue: {
+        get: function () {return this._maxValue},
+        set: function (value) {
+            this._maxValue = value;
+        }
+    },
+
+    /**
+     * Orientation of the bar will be horizontal by default.
+     * The orientation is set to automatically adjust according to the value
+     * of the screen width.
+     */
+    orientation: {value: null},
+    /**
+     * Allows user to override orientation.
+     * TODO: needed?
+     */
+    orientationOverride: {value: null},
+    /**
+     * TODO: needed?
+     */
```

Comment:
It would be simpler if the API to set the orientation would be the "orientation" property and if there were no "orientationOverride" property.